### PR TITLE
Code-compiler & index-manager support for compound type

### PR DESCRIPTION
**Pull request by thala321**
- Code-compiler support adding schema file (**done**)
- Support compound type:
  - add plugin migration (**done**) 
  - add plugin hasura (**done**) 
- Refactor index-manager-helper to:
  - builder pattern for index-config (**done**)
  - move chain_reader_client to a separate file (**done**)
  - move config parser to a separate file (**done**)
  - move manifest parser to a separate file (**done**)
- Remove the usage of query, and table.txt and use schema.graphql (**done**)
- Rebase new code for plugin migration and plugin hasura from anh Tai (make a new MR for this)

### CODE REVIEW

File: indexer/index-manager/src/store.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
+use diesel::{PgConnection, RunQueryDsl};
+use std::process::Command;
+use std::fs::File;
+use std::io::Read;
+use lazy_static::lazy_static;
+
+lazy_static! {
+    static ref INDEXER_MIGRATION_FILE: String = String::from("./indexer/migration/indexers.sql");
```
Comment (daywednes) [inline indexer/index-manager/src/store.rs:8]: why hard-code?

**Comment by thala321** [comment]
Please squash merge this with the following comment:

- Code-compiler and Index Manager support adding schema file
- Code-compiler and Index Manager support compound type
- Refactor index-manager-helper

**Review by daywednes — COMMENTED** [review]
this PR is too big - let's review and merge as long as the logic works as expected. Minor issues can be improved in the next PR

File: indexer/index-manager/src/store.rs (hunk 1 above)
Comment (thala321) [reply indexer/index-manager/src/store.rs:8]: this table is to store indexer list, not indexer detail

File: code-compiler/app.py [hunk 2]
```diff
@@ -107,6 +107,7 @@ def compile_handler():
     project = urllib.parse.unquote_plus(data["project.yaml"])
     up = urllib.parse.unquote_plus(data["up.sql"])
```
Comment (anhhuy0501) [inline code-compiler/app.py:108]: remove up.sql

File: code-compiler/app.py [hunk 3]
```diff
@@ -107,6 +107,7 @@ def compile_handler():
...
     table = urllib.parse.unquote_plus(data["table"])  # Should refactor this to index_name in the future
```
Comment (anhhuy0501) [inline code-compiler/app.py:109]: Remove table

File: indexer/example/schema.graphql [hunk 4]
```diff
@@ -0,0 +1 @@
+abc local
```
Comment (anhhuy0501) [inline indexer/example/schema.graphql:1]: remove support for loading from local files

File: indexer/index-manager/src/core_helper.rs [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        .unwrap_or(String::from("postgres://graph-node:let-me-in@localhost"));
+    static ref IPFS_ADDRESS: String =
+        env::var("IPFS_ADDRESS").unwrap_or(String::from("0.0.0.0:5001"));
+}
+
+pub async fn loop_blocks(params: DeployParams) -> Result<(), Box<dyn Error>> {
+    // Get user index mapping logic, query for migration and index's configurations
+    let index_config = match params.deploy_type {
+        DeployType::Local => {
+            let index_config = IndexConfigLocalBuilder::default()
```
Comment (anhhuy0501) [inline indexer/index-manager/src/core_helper.rs:28]: remove loading local file

File: indexer/index-manager/src/chain_reader_client.rs [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+        end_block_number: 1,
+        chain_type: chain_type as i32,
+    };
+    let mut stream = client
+        .list_blocks(Request::new(get_blocks_request))
+        .await?
+        .into_inner();
+    // Subscribe new blocks
+    log::info!("[Index Manager Helper] Start processing block");
+    while let Some(data) = stream.message().await? {
```
Comment (anhhuy0501) [inline indexer/index-manager/src/chain_reader_client.rs:40]: Todo (later): refactor to use function from chain-reader.

File: indexer/index-manager/src/config.rs [hunk 7]
```diff
@@ -0,0 +1,172 @@
+use ipfs_client::core::create_ipfs_clients;
```
Comment (anhhuy0501) [inline indexer/index-manager/src/config.rs:1]: add comment to describe the function

**Review by anhhuy0501 — APPROVED** [review]
LGTM

**Pull request merged**